### DISCUSSION TITLE
[Bug?] Get shipping rates

### DISCUSSION THREAD

**Discussion by josias-hosang** [swellstores/community#392] [Help] [answered]
On my swell instance merz (test mode) I experience some problems with getting the shipping rates. It's a simple configuration with one zone and shipment rule. With frontend API adding an item and setting country, the `swell.cart.getShippingRates()` gets me no service. So I tried for easier demonstration a draft order in backend. And there is the same problem. I double checked and reset configuration; I also have an other instance (skipp-test) where the same configuration is working as it should. See attached screenshots:

Settings:
<img width="542" alt="settings-shipping" src="https://github.com/swellstores/community/assets/48205514/c43f6adc-913c-4dfc-b4ab-a45552bf7bc1">

Draft order:
<img width="714" alt="draft-order" src="https://github.com/swellstores/community/assets/48205514/b98550b5-bbdb-49af-8f80-d5f1194acc44">

Edit shipping (the "Refresh rate" button results in empty price):
<img width="370" alt="edit-shipping" src="https://github.com/swellstores/community/assets/48205514/84cfa9aa-528c-4cc3-8d21-3138c1aa449f">

Can you help me with this? Thanks.

**Comment by josias-hosang** [comment, marked as answer]
Found it by myself. In /settings/shipments (and also in my service) currency was set to USD. You can't change/see it in the dashboard. Maybe this could be added? Changing and rechanging currency in the general settings didn't affect shipment currency. I changed it via console and now it's working again.

**Reply by SusannaSwell** [reply]
hello @josias-hosang thank you for pointing this, I will create a ticket in our internal system for fixing this

**Reply by Kealtie** [reply]
Can you share the steps you used to change this through the console, I think this is also my issue.

**Comment by ItsAndrewM** [comment]
Having the same issue while creating a custom checkout.  I tried to run `swell.cart.getShippingRates()` but keep getting null in return.  My shipping settings mirror Josias's.  Does each shipment service need to be configured for each currency supported?

**Reply by coderreco** [reply]
Make sure you have a country code added to the cart—somehow I missed that I didn't when I was testing and chased a non-existent issue for a couple of hours. Won't return a rating if it is missing.

**Reply by ItsAndrewM** [reply]
I'm still kind of stuck on this.  Currently, the cart has a shipping.country value ("CA") but I'm still getting null.  I appreciate the help though
